### DISCUSSION TITLE
Can mkcert specify "issued to" when generating certificates?

### DISCUSSION THREAD

**Discussion by isboy88** [FiloSottile/mkcert#598] [Q&A] [answered]
Can mkcert specify "issued to" when generating certificates? "issued to" is not specified in the certificate, and Synology DSM will display blank after importing the certificate.

![image](https://github.com/FiloSottile/mkcert/assets/19484319/492b6b7a-e498-474d-8443-60c4534c539e)

**Comment by isboy88** [comment, marked as answer]
add CN to generated certificates · elexx/mkcert@49fec4a
https://github.com/elexx/mkcert/commit/49fec4a21908ddf9cb3600fed28ea9c6de8bca93

**Comment by DoraTheExploder** [comment]
I've discovered that if you use an external tool to generate a Certificate Signing Request (.csr) such as [DigiCert CSR Tool](https://www.digicert.com/easy-csr/openssl.htm) (or do it manually with openssl), you can customize the relevant fields and then pass it as a sort of configuration file to mkcert with `$ mkcert -csr <domain>`.

I have not tested using a CSR for initial Root CA installation, but `$ mkcert --help` implies that it should be possible.
